packages/sdk/src/oracle/client.ts

```diff
@@ -1,23 +1,102 @@
+        "Must specify user address as checkum address"
+      );
+    return this.sm.types.setUser.create(params);
   }
-  setActiveRequest(params: Inputs["request"]): string {
+  clearUser(): string {
+    return this.sm.types.clearUser.create(undefined);
+  }
+  setActiveRequest(params: InputRequest): string {
     return this.sm.types.setActiveRequest.create(params);
   }
+  approveCollateral(): string {
+    const request = this.store.read().request();
+    const inputRequest = this.store.read().inputRequest();
+    const user = this.store.read().user();
+    const oracleAddress = this.store.read().oracleAddress();
+    assert(user.address, "requires a user account address");
+    assert(user.signer, "requires a user signer");
+    assert(user.chainId === inputRequest.chainId, "On wrong chain");
+    return this.sm.types.approve.create(
+      {
+        currency: request.currency,
+        account: user.address,
+        chainId: inputRequest.chainId,
+        signer: user.signer,
+        spender: oracleAddress,
+        amount: ethers.constants.MaxUint256.toString(),
+        confirmations: 1,
+      },
+      user.address
+    );
+  }
+  proposePrice(proposedPriceDecimals: string | number): string {
+    const proposedPrice = toWei(proposedPriceDecimals);
+    const inputRequest = this.store.read().inputRequest();
+    const request = this.store.read().request();
+    const user = this.store.read().user();
+    assert(user.address, "requires a user account address");
+    assert(user.signer, "requires a user signer");
+    assert(user.chainId === inputRequest.chainId, "On wrong chain");
+    return this.sm.types.proposePrice.create(
+      {
+        ...inputRequest,
+        proposedPrice,
+        signer: user.signer,
+        account: user.address,
+        currency: request.currency,
+        confirmations: 1,
+      },
+      user.address
+    );
+  }
+  disputePrice(): string {
+    const inputRequest = this.store.read().inputRequest();
+    const user = this.store.read().user();
+    const request = this.store.read().request();
+    assert(user.address, "requires a user account address");
+    assert(user.signer, "requires a user signer");
+    assert(user.chainId === inputRequest.chainId, "On wrong chain");
+    return this.sm.types.disputePrice.create(
+      {
+        ...inputRequest,
+        confirmations: 1,
+        signer: user.signer,
+        account: user.address,
+        currency: request.currency,
+      },
+      user.address
+    );
+  }
+  switchOrAddChain(): string {
+    const inputRequest = this.store.read().inputRequest();
+    const user = this.store.read().user();
+    assert(user.provider, "requires user provider");
+    assert(user.address, "requires user address");
+    assert(inputRequest.chainId, "requires active request chainId");
+    return this.sm.types.switchOrAddChain.create(
+      { chainId: inputRequest.chainId, provider: user.provider },
+      user.address
+    );
+  }
```

Comment:
all these functions create state machine calls. see the handler definitions in the files for business logic